homeassistant/components/aprilaire/climate.py

```diff
@@ -298,3 +361,153 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
             raise ValueError(f"Unsupported preset mode {preset_mode}")
 
         await self.coordinator.client.read_scheduling()
+
+    async def async_set_dehumidity(self, dehumidity: int) -> None:
+        """Set the target dehumidification setpoint."""
+
+        if self.coordinator.data.get(Attribute.DEHUMIDIFICATION_AVAILABLE) == 1:
+            await self.coordinator.client.set_dehumidification_setpoint(dehumidity)
+        else:
+            raise ValueError(
+                "Device does not support setting dehumidification setpoint"
+            )
+
+    async def async_trigger_air_cleaning_event(self, event: str) -> None:
+        """Triggers an air cleaning event of 3 or 24 hours."""
+
+        if self.coordinator.data.get(Attribute.AIR_CLEANING_AVAILABLE) == 1:
+            current_air_cleaning_mode = self.coordinator.data.get(
+                Attribute.AIR_CLEANING_MODE, 0
+            )
+
+            if event == "3hour":
+                await self.coordinator.client.set_air_cleaning(
+                    current_air_cleaning_mode, 3
+                )
+            elif event == "24hour":
+                await self.coordinator.client.set_air_cleaning(
+                    current_air_cleaning_mode, 4
+                )
+            else:
+                raise ValueError(f"Invalid event {event}")
```

Comment:
We should raise `ServiceValidationError` instead, and preferably we also add a translation key to make all the errors translatable :)